### DISCUSSION TITLE
How do I tell Cypress runner to ignore temporary files?

### DISCUSSION THREAD

**Discussion by cjcrandall** [cypress-io/cypress#14243] [Questions and Help]
When I have Cypress running (via `cypress open`), if I try to edit a test file in emacs, a temporary file prefixed with `.#` gets created. This immediately crashes the browser window running a test and I get the following error:

**Title:** Error running plugin

**Message:** The following error was thrown by a plugin. We stopped running your tests because a plugin crashed. Please check your plugins file (`/<redacted>/cypress/plugins/index.js`)

**Details:** Error: ENOENT: no such file or directory, stat '/\<redacted\>/cypress/integration/.#test-name.e2e-spec.ts'

**Stack trace:**
```
Error: The following error was thrown by a plugin. We stopped running your tests because a plugin crashed. Please check your plugins file (`/<redacted>/cypress/plugins/index.js`)
    at Object.get (/home/<redacted>/.cache/Cypress/5.5.0/Cypress/resources/app/packages/server/lib/errors.js:968:15)
    at EventEmitter.handleError (/home/<redacted>/.cache/Cypress/5.5.0/Cypress/resources/app/packages/server/lib/plugins/index.js:159:20)
    at EventEmitter.emit (events.js:310:20)
    at ChildProcess.<anonymous> (/home/<redacted>/.cache/Cypress/5.5.0/Cypress/resources/app/packages/server/lib/plugins/util.js:19:22)
    at ChildProcess.emit (events.js:310:20)
    at emit (internal/child_process.js:876:12)
    at processTicksAndRejections (internal/process/task_queues.js:85:21)

```

Possibly relevant: I'm using TypeScript. Any ideas whether there's something I should configure in my `tsconfig` file or plugins file, or is this a Cypress bug? I don't remember running into this before. I'm on Cypress 5.5.0 on Ubuntu 20.04.

**Comment by bahmutov** [comment]
You can set the pattern of files to ignore using ignoreTestFiles setting, see http://on.cypress.io/configuration

**Reply by cjcrandall** [reply]
Looking at this more closely, I think there's something else going on that may be a bug. We recently transitioned from an internal npm server to GitHub for storing our javascript packages, and in the process of upgrading I deleted my repository's `yarn.lock` file. A number of packages updated to newer versions when I reran `yarn install`, and I found that this behavior started after that commit in my repo.

I did experiment a bit with `ignoreTestFiles` but couldn't make the error go away. Hopefully I can track it down enough to provide a reproducible example and enter an issue--in the meantime this is incredibly annoying as I'm trying to edit tests 🙁.

**Reply by cjcrandall** [reply]
I thought a likely suspect was `@cypress/webpack-preprocessor` updating from 4.1.1 to 4.1.5, but reverting that change didn't make a difference. Unfortunately, my lock file is over 6,000 lines long so I'm not very optimistic about tracking the problem down by scrolling through the diff. Gleb, would you have a short list of other "likely suspect" dependencies related to file watching that I should specifically check?

**Reply by cjcrandall** [reply]
It's also worth noting that these emacs temporary files do _not_ show up in the list of spec files in the `</> Tests` pane in Cypress. The crash happens a few seconds after I click a spec file to run, after Chrome shows "Your tests are loading" but before the tests start running (or pretty immediately if I modify a file while the tests are running).

**Reply by cjcrandall** [reply]
I just noticed the following, slightly different error information getting printed to my terminal as well when this happens:

```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at process.target._send (internal/child_process.js:832:20)
    at process.target.send (internal/child_process.js:703:19)
    at callback (/<redacted>/node_modules/worker-farm/lib/child/index.js:32:17)
    at module.exports (/<redacted>/node_modules/terser-webpack-plugin/dist/worker.js:13:5)
    at handle (/<redacted>/worker-farm/lib/child/index.js:44:8)
    at process.<anonymous> (/<redacted>/node_modules/worker-farm/lib/child/index.js:55:3)
    at process.emit (events.js:315:20)
    at emit (internal/child_process.js:903:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
Emitted 'error' event on process instance at:
    at internal/child_process.js:836:39
    at processTicksAndRejections (internal/process/task_queues.js:75:11) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

I also noticed that `terser-webpack-plugin` updated from 1.4.3 to 1.4.5, which included an update of one of its dependencies, `serialize-javascript`, from 2.1.2 to 4.0.0. I jumped at that and thought it could be a "smoking gun" but hand-editing my `yarn.lock` file to back off that change didn't change the behavior. Neither did updating my `package.json` file to remove all the `@`, `~`, or `^` prefixes to try to install exact, older versions of everything. I feel like I'm missing something obvious...

**Reply by cjcrandall** [reply]
By reverting to my `yarn.lock` file prior to the change, I was able to narrow this down to a `webpack` upgrade from 4.41.6 to 4.44.2. Upgrading to the latest `@cypress/webpack-preprocessor` didn't help, though I also haven't tried upgrading from Cypress 5.5.0 to the latest version. For now, I went back to my original lock file, used `sed` to find and replace all our internal server references with `registry.npmjs.org`, and got things working again.

**Comment by peey** [comment]
This might be related to your issue: https://github.com/cypress-io/cypress/issues/8759
